inception-concept-linking/src/main/java/de/tudarmstadt/ukp/inception/conceptlinking/recommender/NamedEntityLinker.java

```
@@ -18,213 +18,223 @@
+
+        Feature labelFeature = predictionType.getFeatureByBaseName("label");
+
+        for (KBHandle prediction : handles.stream().limit(numPredictions)
+            .collect(Collectors.toList())) {
+            AnnotationFS annotation = aJcas.getCas().createAnnotation(predictionType, aBegin, aEnd);
+            annotation.setStringValue(labelFeature, prediction.getIdentifier());
+            aJcas.getCas().addFsToIndexes(annotation);
+        }
+
     }
 
-    private boolean isNamedEntity(TokenObject token)
+    private List<KBHandle> readCandidates(KnowledgeBase kb, String aCoveredText, int aBegin,
+        JCas aJcas)
     {
-        return nerAnnotations.stream()
-            .map(AnnotationObject::getTokenObject)
-            .anyMatch(t -> t.getOffset().equals(token.getOffset())
-                   && t.getDocumentURI().equals(token.getDocumentURI()));
+        return kbService
+            .read(kb, (conn) -> clService.disambiguate(kb, null, aCoveredText, aBegin, aJcas));
     }
 
-    private List<KBHandle> readCandidates(KnowledgeBase kb, TokenObject token) {
-        return kbService.read(kb, (conn) -> {
-            SourceDocument doc = documentService
-                .getSourceDocument(project, token.getDocumentName());
-            AnnotationDocument annoDoc = documentService
-                .createOrGetAnnotationDocument(doc, user);
-            JCas jCas;
-            try {
-                jCas = documentService.readAnnotationCas(annoDoc);
-                return clService.disambiguate(kb, "", token.getCoveredText(),
-                    token.getOffset().getBeginCharacter(), jCas);
-            }
-            catch (IOException e) {
-                log.error("An error occurred while retrieving entity candidates.", e);
-                return Collections.emptyList();
-            }
-        });
+    @Override
+    public double evaluate(List<CAS> aCasses, DataSplitter aDataSplitter)
+    {
+        return -1;
```

Should throw an exception when called, e.g. `UnsupportedOperationException`.